storagetransfer/conftest.py

```diff
@@ -32,30 +32,76 @@ def project_id():
     yield os.environ.get("GOOGLE_CLOUD_PROJECT")
 
 
-@pytest.fixture(scope='module')
+# cache secret from secret manager
+aws_secret_manager_cache = None
+
+
+def aws_retrieve_from_secret_manager(name: str):
+    """
+    Retrieve a secret given a name.
+
+    example ``name`` = ``projects/123/secrets/my-secret/versions/latest``
+    """
+
+    global aws_secret_manager_cache
+
+    if aws_secret_manager_cache:
+        return aws_secret_manager_cache
+
+    client = secretmanager.SecretManagerServiceClient()
+
+    # retrieve from secret manager
+    response = client.access_secret_version(request={"name": name})
+
+    # parse secret from secret manager
+    secret = json.loads(response.payload.data.decode("UTF-8"))
+
+    aws_secret_manager_cache = {
+        'aws_access_key_id': secret['AccessKeyId'],
+        'aws_secret_access_key': secret['SecretAccessKey']
+    }
+
+    return aws_secret_manager_cache
+
+
+def aws_key_pair():
+    sts_aws_secret_name = os.environ.get("STS_AWS_SECRET_NAME")
+    aws_access_key_id = os.environ.get("AWS_ACCESS_KEY_ID")
+    aws_secret_access_key = os.environ.get("AWS_SECRET_ACCESS_KEY")
+
+    if sts_aws_secret_name:
+        return aws_retrieve_from_secret_manager(sts_aws_secret_name)
+    else:
+        return {
+            'aws_access_key_id': aws_access_key_id,
+            'aws_secret_access_key': aws_secret_access_key
+        }
```

Comment:
`else` in this case is optional, so not having it reduces nesting and still remains straightforward code.
```suggestion
    
    return {
        'aws_access_key_id': aws_access_key_id,
        'aws_secret_access_key': aws_secret_access_key
    }
```

Reply:
Thanks, fixing